lib/pages/caregiver/plan_details_page.dart

```diff
@@ -25,141 +26,118 @@ class CaregiverPlanDetailsPage extends StatefulWidget {
 
 class _CaregiverPlanDetailsPageState extends State<CaregiverPlanDetailsPage> {
 	static const String _pageKey = 'page.caregiverSection.planDetails';
-	Mongo.ObjectId planID;
+
 
   @override
   Widget build(BuildContext context) {
-		Mongo.ObjectId planID = ModalRoute.of(context).settings.arguments;
-
 		return Scaffold(
 			body: Column(
 				crossAxisAlignment: CrossAxisAlignment.start,
 				children: [
-					AppHeader.widget(
-						title: '$_pageKey.header.title',
-						appHeaderWidget: ItemCard(
-							title: planID != null ? planID.toString() : "Sprzątanie pokoju",
-							subtitle: "Co każdy poniedziałek, środę, czwartek i piątek",
-							chips:
-							<Widget>[
-								AttributeChip.withIcon(
-									content: AppLocales.of(context).translate('page.caregiverSection.plans.content.tasks', {'NUM_TASKS': 1}),
-									color: Colors.indigo,
-									icon: Icons.layers
-								)
-							],
-						),
-						helpPage: 'plan_info',
-						popupMenuWidget: PopupMenuList(
-							lightTheme: true,
-							items: [
-								UIButton.ofType(ButtonType.edit, () => Navigator.of(context).pushNamed(AppPage.caregiverPlanForm.name, arguments: Mongo.ObjectId.parse('5f3ad3cf4ea005fba95da86b'))),
-								UIButton.ofType(ButtonType.delete, () => showBasicDialog(
-									context,
-									GeneralDialog.confirm(
-										title: AppLocales.of(context).translate('alert.deletePlan'),
-										content: AppLocales.of(context).translate('alert.confirmPlanDeletion'),
-										confirmText: 'actions.delete',
-										confirmAction: () => Navigator.of(context).pop(),
-										confirmColor: Colors.red
-									)
-								))
-							],
-						)
-					),
-					AppSegments(segments: _buildPanelSegments())
+					LoadableBlocBuilder<CaregiverTasksCubit>(
+						builder: (context, state) => AppSegments(segments: _buildPanelSegments(state)),
+					)
 				],
 			),
 		);
 
 	}
 
-	List<Segment> _buildPanelSegments() {
+	List<Widget> _buildPanelSegments(CaregiverTasksLoadSuccess state) {
+  	UIPlan uiPlan = state.uiPlan;
+  	List<UITask> mandatoryTasks = state.tasks.where((task) => task.optional == false).toList();
+		List<UITask> optionalTasks = state.tasks.where((task) => task.optional == true).toList();
+
 		return [
+			_getCardHeader(uiPlan),
 			_getTasksSegment(
```

Comment:
Those two task segment functions could easily be a single one with some more parameters - the widget tree is identical